### PR TITLE
Remove useless variable decl in `into_bigint` Mont

### PR DESCRIPTION
**Pull request by tcoratger**
In this pull request, it is proposed to streamline the code by eliminating the unnecessary `tmp` variable. 

Currently, the `tmp` variable serves the sole purpose of holding the `[u64; N]` array temporarily to store the final `BigInt` returned at the end of the function. 

However, this approach can be made more readable and concise. Therefore, I suggest removing the `tmp` variable and replacing it with a direct declaration of `BigInt` at the function's conclusion. This modification enhances code clarity and removes redundant assignments.

### CODE REVIEW

File: ff/src/fields/models/fp/montgomery_backend.rs [hunk 1]
```diff
@@ -371,8 +371,7 @@ pub trait MontConfig<const N: usize>: 'static + Sync + Send + Sized {
     #[unroll_for_loops(12)]
     #[allow(clippy::modulo_one)]
     fn into_bigint(a: Fp<MontBackend<Self, N>, N>) -> BigInt<N> {
-        let mut tmp = a.0;
-        let mut r = tmp.0;
+        let mut r = a.0 .0;
```
Comment (mmaker) [inline ff/src/fields/models/fp/montgomery_backend.rs:374]: why the whitespace?

Comment (Pratyush) [reply ff/src/fields/models/fp/montgomery_backend.rs:374]: the whitespace is there for the same reason we had the variable: rust thinks `a.0.0` is actually `a.(0.0)`, and parse the latter part as a float

**Review by Pratyush — APPROVED** [review]

File: ff/src/fields/models/fp/montgomery_backend.rs (hunk 1 above)
Comment (mmaker) [reply ff/src/fields/models/fp/montgomery_backend.rs]: gosh

Comment (tcoratger) [reply ff/src/fields/models/fp/montgomery_backend.rs]: I think it's rust fmt stuff. I found it strange at the beginning but I found a lots of related issues in [rustfmt](https://github.com/rust-lang/rustfmt) repo. Like this one: https://github.com/rust-lang/rustfmt/issues/5745

Comment (mmaker) [reply ff/src/fields/models/fp/montgomery_backend.rs]: this seems to work on play.rustlang tho
```

struct B (pub u8, pub u8);
struct A (pub B, pub B);

fn main() { 
    let a = A(B(0, 1), B(2, 3));
    println!("{}", a.0.0)
}
```

**Pull request merged**